extensions/elytron-security/deployment/src/main/java/io/quarkus/elytron/security/deployment/SecurityDeploymentProcessor.java

```diff
@@ -176,11 +178,46 @@ AuthConfigBuildItem configureMPRealmConfig(SecurityRecorder recorder,
         return null;
     }
 
+    /**
+     * Check to see if a JdbcRealmConfig was specified and enabled and create a
+     * {@linkplain org.wildfly.security.auth.realm.LegacyPropertiesSecurityRealm}
+     * runtime value to process the user/roles properties files. This also registers the names of the user/roles properties
+     * files
+     * to include the build artifact.
+     *
+     * @param recorder - runtime security recorder
+     * @param securityRealm - the producer factory for the SecurityRealmBuildItem
+     * @param passwordRealm - the producer factory for the PasswordRealmBuildItem
+     * @param beanContainer - the CDI bean container
+     * @param dataSourceInitialized Added this dependency to be sure that Agroal is initialized first
+     * @return the AuthConfigBuildItem for the realm authentication mechanism if there was an enabled JdbcRealmConfig,
+     *         null otherwise
+     * @throws Exception - on any failure
+     */
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    AuthConfigBuildItem configureJdbcRealmAuthConfig(SecurityRecorder recorder,
+            BuildProducer<SecurityRealmBuildItem> securityRealm,
+            BuildProducer<PasswordRealmBuildItem> passwordRealm,
+            BeanContainerBuildItem beanContainer,
+            DataSourceInitializedBuildItem dataSourceInitialized) throws Exception {
```

Comment:
Hi, I'm not sure the agroal code has to be referenced in. What about the idea discussed earlier where you have `DataSource` produced by Agroal injected ? Only the integration tests module should include a non-optional agroal dependency, this module should not, unless I'm missing something :-)

Reply:
🤔 Looking at the Flyway extension which relies use a CDI injection to get the Datasource instance, the `deployment` module as a dependency on `DataSourceInitializedBuildItem dataSourceInitialized`.  
https://github.com/quarkusio/quarkus/blob/master/extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

My comprehension is this build item ensures you have an instance of a `DataSource` before invoking the Producer

I'm not sure to understand how you want to get rid off the `agroal-deployment` dependency...

Reply:
@danielpetisme If I understand it correctly, in order to be able to access an instance of `DataSource` inside the `SecurityRecorder` you must add the `DataSourceInitializedBuildItem dataSourceInitialized` or there will not be any datasource yet in the bean factory.

So this is a build order issue ... and I'm afraid I don't know if there is a way to give a hint to the current build algorithm to execute this step after an other one.

If nobody find better solutions, I found three that all have some caveats:

- Separate this in a new extension to be able to have a _hard dependency_ to Agroal
- Provide the DataSource lazily (with some kind of provider or proxy): maybe difficult to build
- Move the `DataSourceInitializedBuildItem` to `quarkus-core-deployment`: the simplest one :)

Maybe you can use the third one for the moment to be able to easily finish your first implementation and discuss this point later on ?

Reply:
Oh, sorry, I missed that this is how the flyway processor does it. So please continue as you agreed with Loic. 
My current concern is that this security module is not only about the DB, while Flyway is, so I was hoping that simply adding the agroal dependency to the user application only which needs a DB security storage would do, but I do not know now how else to enforce that the agroal is initialized first :-), let me actually ask, I think it can be of general interest. Some work around `io.quarkus.deployment.Capabilities` might need to be done. But in meantime, please continue as you plan, thanks

Reply:
I propose to stick with the Agroal dependency set in the `elytron-security` extension to move on.
IMHO, a dedicated `elytron-security-jdbc` extension would be the easiest and the more explicit. The dev wouldn't have to do any further configuration.

I propose to discuss this point on zulip?

Reply:
@danielpetisme yes, maybe some people will have different solution on Zulip :) 
Please include my three proposals on your message on Zulip as I don't go there very often.

Reply:
https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/quarkus-elytron-security/near/174217865